filtering/lexer.go

```diff
@@ -123,7 +123,10 @@ func (l *Lexer) nextRune() (rune, error) {
 	switch {
 	case n == 0:
 		return r, io.EOF
-	case r == utf8.RuneError:
+	// Not a valid UTF-8 rune, but not the replacement character.
+	case r == utf8.RuneError && (n != 3 || l.remainingFilter()[:3] != "\xef\xbf\xbd"):
```

Comment:
Lets expand the comment here a bit, to make the distinction more clear, something like:
```
// If the input rune was the replacement character (`\uFFFD`) preserve it. 
// If the input rune was invalid (and converted to replacement character) return an error.
```

Reply:
Done.